### PR TITLE
[FX-5808] Fix badge placement in tabs and checkbox spacing

### PR DESCRIPTION
**Pull request by sashuk**
[FX-5808]

### Description

This pull request

- removes `top-0 right-0` classes from Badge component.

[Original MUIv4 implementation](https://mui.com/base-ui/react-badge/components-api/) has these styles only for the badge (not root) itself in https://github.com/mui/material-ui/blob/v4.x/packages/material-ui/src/Badge/Badge.js#L78 (particularly `anchorOriginTopRightRectangular` as this is the only way Picasso used to use Badge).

- restores original font size for Checkbox root

### How to test

- [Temploy](https://picasso.toptal.net/fx-5808-badges-are-misplaced)
- Check how tabs look in Staff Portal in https://t-staff-portal-pr-13553.toptal.net/clients/160751
- Check visual changes https://happo.io/a/675/p/1189/compare/b6f2db484bda3bfb5ffd325f92aeb174639783ab/187a936cb83477f540b33d1e2cd57cbb021a90f1?t=added

### Screenshots

**Tabs**

Before https://staff-portal.toptal.net/clients/160751

Screenshot 2024-08-19 at 17 33 54

After https://t-staff-portal-pr-13553.toptal.net/clients/160751

Screenshot 2024-08-19 at 17 34 14

Checkbox

Before 

Screenshot 2024-08-20 at 12 16 28

After https://t-staff-portal-pr-13553.toptal.net/sourcing_requests/1515/update

Screenshot 2024-08-20 at 12 33 08

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Ensure that deployed demo has expected results and good examples
- [N/A] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-5808]: https://toptal-core.atlassian.net/browse/FX-5808?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Comment by sashuk** [comment]
@toptal-anvil ping reviewers

File: packages/base/Badge/src/Badge/Badge.tsx [hunk 1]
```diff
@@ -58,8 +58,7 @@ export const Badge = forwardRef<HTMLDivElement, Props>(function Badge(
         root: {
           className: twMerge(
             `inline-flex flex-shrink-0 content-middle flex-nowrap justify-normal 
-            text-[10px] font-semibold leading-3 align-middle text-graphite-700 
-            top-0 right-0`,
```
Comment (dmaklygin) [inline packages/base/Badge/src/Badge/Badge.tsx:62]: seems to be a leftover: https://github.com/toptal/picasso/commit/6716e0bb3178a7f452f2c79ce56dd524e9bd8685#diff-9b67b829942d99592e9f9827a02960c0056f09f054389bbe6c0d8d1ba93f3f07R62

**Review by dmaklygin — APPROVED** [review]

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Review by ruslan-sed — APPROVED** [review]

**Pull request merged**